src/de.rs

```diff
@@ -240,30 +242,42 @@ fn parse_char_escape_sequence(pair: &Pair<'_, Rule>) -> String {
     })
 }
 
-fn parse_number(pair: &Pair<'_, Rule>) -> f64 {
+fn parse_number(pair: &Pair<'_, Rule>) -> Result<f64> {
+    println!("{:?}", pair);
     match pair.as_str() {
-        "Infinity" => f64::INFINITY,
-        "-Infinity" => f64::NEG_INFINITY,
-        "NaN" | "-NaN" => f64::NAN,
-        s if is_hex_literal(s) => f64::from(parse_hex(&s[2..])),
-        s => s.parse().unwrap(),
+        "Infinity" => Ok(f64::INFINITY),
+        "-Infinity" => Ok(f64::NEG_INFINITY),
+        "NaN" | "-NaN" => Ok(f64::NAN),
+        s if is_hex_literal(s) => Ok(f64::from(parse_hex(&s[2..])?)),
+        s => {
+            if let Ok(r) = s.parse::<f64>() {
+                if r.is_finite() {
+                    Ok(r)
+                } else {
+                    Err(de::Error::custom("error parsing number: too large"))
+                }
+            } else {
+                Err(de::Error::custom("error parsing number"))
```

Comment:
Did you manage to cause this to happen? I have a feeling that this should be unreachable because everything matched by the `decimal_literal` part of the grammar should be a valid Rust literal... No harm in it being here though.

Reply:
tl;dr: no, but just in case I left this in.

Looks like the only way to get this error is to feed invalid number to .parse(), but pest validates the syntax. Basically, I went after all instances of .unwrap() and added error handing there.